_posts/2018-04-09-middleware-event.md

```diff
@@ -0,0 +1,191 @@
+---
+title: Manipulate events using a middleware
+author: Georgos Landin Xanthopoulos
+date: '2018-04-09 13:39:40 +0200'
+header:
+  teaser: images/posts/events.jpg
+  caption: "Photo credit: [**Flickr**](https://www.flickr.com/photos/jone_vasaitis/16022260923/in/photolist-qpQgXZ-asu2YH-63NQ3W-85Gk4g-aEUFCZ-4YvJCc-7pTvgs-7LmUC9-eHjnFV-8E6J8V-hXw713-5DudKc-4VxoHr-9PD8vo-8FHkig-bAACjP-6tpDjd-ab2WT-54tSFr-pqLvEd-dL7Ksg-24g6kGj-ov9pFN-aBQ4zJ-8byiTm-atRt3o-ygkPwT-6SfG7U-24BjkTY-WyZ1Jx-ejbULs-8HjSq7-65nqnz-6G5nvX-22K5Rei-yjR3d-sDFU8-abosaA-cSdMUo-vCuvY-4jKtyC-22wib-cd2Bm3-FkXEeW-6MucmN-dd9ak6-78nBuc-SFNBS8-YYf8G3-5RLXMQ)"
+categories:
+- Symfony
+---
+
+This post will cover how you can add additional data to your events using a SimpleBus middleware. 
+
+Using SimpleBus in your hexagonal architecture is great, due to the versatility it gives you when working with events/commands. 
+If you would like to read more about the event bus or the command bus and their associated handlers/subscribers, see 
+the [documentation](http://docs.simplebus.io/en/latest/index.html). 
+Maybe you want to log something specific in your system, or maybe you are building a gamification application that gives
+users experience points based on their actions. Then using SimpleBus, together with your events and commands might be just
+right for you. 
+
+## How it works
+The main point is to do some kind event manipulation on those events that have already been dispatched inside your application. 
+Maybe you want to save the events to a log, or maybe add some more data to the events before passing them on to another 
+application.
+Below is a section of the middleware class, the entire middleware is shown at the bottom of this post. 
+
+{% highlight php %}
+$eventData = [
+    // .. Add some event data
+]
+// Add the data to a DataContainer object
+$container = new DataContainer($eventData);
+
+$this->eventDispatcher->dispatch('foo', new Foo($event, $container));
+{% endhighlight %}
+
+### The dispatched event
+The Foo event class uses the DataContainer object instead of a
+simple array, since an array is passed by value and not by reference. Therefore by using the DataContainer object we can 
+make sure that the passed parameter has the correct value, since objects are passed by reference, which means that if the 
+variable gets modified, the value will be updated for both parties using it. 
```

Comment:
You should read the paragraph above and correct it for errors. Make sure it is a smooth read.